provider/core/src/erased.rs

```diff
@@ -79,23 +79,12 @@ where
     /// `Yoke` (i.e., `Rc::from(yoke)`).
     fn upcast(other: DataPayload<'static, M>) -> DataPayload<'static, ErasedDataStructMarker> {
         use crate::data_provider::DataPayloadInner::*;
-        match other.inner {
```

Comment:
Nothing changed in this function; I'm only simplifying it to look more like the SerdeSe version.